api/src/main/java/org/infinispan/api/async/AsyncCache.java

```diff
@@ -45,28 +47,84 @@
     */
    CompletionStage<V> get(K key);
```

Comment:
not sure how good idea it is:
```suggestion
   default CompletionStage<V> get(K key) {
      return getEntry(key).thenApply(CacheEntry::value);
   }
```